packages/router/router.ts

```diff
@@ -1026,8 +1026,11 @@ export function createRouter(init: RouterInit): Router {
 
     // On a successful navigation we can assume we got through all blockers
     // so we can start fresh
-    let blockers = new Map();
-    blockerFunctions.clear();
+    let blockers = state.blockers;
+    if (blockers.size > 0) {
+      blockers = new Map(blockers);
+      blockers.forEach((_, k) => blockers.set(k, IDLE_BLOCKER));
+    }
```

Comment:
Deleting the blockers here was incorrect if they existed in a reused layout route.  Instead we just want to set them back to `IDLE_BLOCKER` and let the `useBlocker` `useEffect` handle deletion on unmount